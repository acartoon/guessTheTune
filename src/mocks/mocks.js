import {getRandomInt, getRandomString} from "../utils";

export const settings = {
  gameTime: 5,
  errorCount: 3,
};

const ANSWERS_COUNT_GENRE = 4;
const ANSWERS_COUNT_ARTIST = 3;
const QUESTION_COUNT = 5;

const GENRES = [`rock`, `pop`, `jazz`];

const ARTISTS = [`John Snow`, `Jack Daniels`, `Jim Beam`];
// const SRC = `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`;
const SRC = `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`;
const IMG_SRC = [`https://api.adorable.io/avatars/102`, `https://api.adorable.io/avatars/120`, `https://api.adorable.io/avatars/134`];

const getQuestionsGenre = () => {
  return {
    id: getRandomString(3),
    type: `genre`,
    genre: GENRES[getRandomInt(GENRES.length - 1)],
    answers: new Array(ANSWERS_COUNT_GENRE).fill(``).map(getAnswersGenre),
  };
};

const getQuestionsArtict = () => {
  return {
    id: getRandomString(3),
    type: `artist`,
    song: {
      artist: ARTISTS[getRandomInt(ARTISTS.length - 1)],
      src: SRC,
    },
    answers: new Array(ANSWERS_COUNT_ARTIST).fill(``).map(getAnswersArtict),
  };
};

const getAnswersGenre = () => {
  return {
    id: getRandomString(3),
    src: SRC,
    genre: GENRES[getRandomInt(GENRES.length - 1)],
  };
};

const getAnswersArtict = () => {
  return {
    id: getRandomString(3),
    picture: IMG_SRC[getRandomInt(IMG_SRC.length - 1)],
    artist: ARTISTS[getRandomInt(ARTISTS.length - 1)],
  };
};


export const questions = new Array(QUESTION_COUNT).fill(``).map(() => {
  return getRandomInt(1) ? getQuestionsArtict() : getQuestionsGenre();
});
