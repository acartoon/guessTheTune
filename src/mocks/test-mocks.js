export const testMocks = [
  {
    answers: [
      {id: `9mT`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `fRl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `LIz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
      {id: `qJ3`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
    ],
    genre: `jazz`,
    id: `7wO`,
    type: `genre`,
  },
  {
    answers: [
      {id: `Sub`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `aWl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `BmX`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `ZGz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
    ],
    genre: `pop`,
    id: `30K`,
    type: `genre`,
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Snow`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Lenon`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
];
