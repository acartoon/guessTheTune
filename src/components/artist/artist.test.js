import React from 'react';
import renderer from 'react-test-renderer';
import Artist from './artist';
import {testMocks} from '../../mocks/test-mocks';


it(`Artist correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<Artist
    onAnswer={() => null}
    question={testMocks[2].song}
    i={154}
  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
