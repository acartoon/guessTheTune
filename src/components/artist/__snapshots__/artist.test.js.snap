// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist correctly renders after relaunch 1`] = `
<div
  className="artist"
>
  <input
    className="artist__input visually-hidden"
    id="answer-154"
    name="answer"
    onChange={[Function]}
    type="radio"
    value="answer-154"
  />
  <label
    className="artist__name"
    htmlFor="answer-154"
  >
    <img
      alt="John Snow"
      className="artist__picture"
    />
    John Snow
  </label>
</div>
`;
